File: front/src/app/global/models/Game.ts
import Editor from "./Editor";
import Category from "./Category";

class Game {

    // Attributes
    private _id: number;
    private _name: string;
    private _image: string;
    private _description: string;
    private _players: number;
    private _duration: string;
    private _published: string;
    private _rating: number;
    private _age: number;
    private _editor: Editor;
    private _categories: Category[];
    private _extensions: any[];

    // Constructor
    constructor(id: number, name: string, image: string, description: string, players: number, 
        duration: string, published: string, rating: number, age: number, 
        editor: Editor, categories: Category[], extensions: any[]) {
        this._id = id;
        this._name = name;
        this._image = image;
        this._description = description;
        this._players = players;
        this._duration = duration;
        this._published = published;
        this._rating = rating;
        this._age = age;
        this._editor = editor;
        this._categories = categories;
        this._extensions = extensions;
    }

    // Getters / Setters
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }

    public get name(): string {
        return this._name;
    }
    public set name(value: string) {
        this._name = value;
    }

    public get image(): string {
        return this._image;
    }
    public set image(value: string) {
        this._image = value;
    }

    public get description(): string {
        return this._description;
    }
    public set description(value: string) {
        this._description = value;
    }

    public get players(): number {
        return this._players;
    }
    public set players(value: number) {
        this._players = value;
    }

    public get duration(): string {
        return this._duration;
    }
    public set duration(value: string) {
        this._duration = value;
    }

    public get published(): string {
        return this._published;
    }
    public set published(value: string) {
        this._published = value;
    }

    public get rating(): number {
        return this._rating;
    }
    public set rating(value: number) {
        this._rating = value;
    }

    public get age(): number {
        return this._age;
    }
    public set age(value: number) {
        this._age = value;
    }

    public get editor(): Editor {
        return this._editor;
    }
    public set editor(value: Editor) {
        this._editor = value;
    }

    public get categories(): Category[] {
        return this._categories;
    }
    public set categories(value: Category[]) {
        this._categories = value;
    }

    public get extensions(): any[] {
        return this._extensions;
    }
    public set extensions(value: any[]) {
        this._extensions = value;
    }
}

export default Game;

File: back/src/main/java/fr/mds/debazar/core/model/Game.java
package fr.mds.debazar.core.model;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name= "GAME")
public class Game {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String name;
    private String image;
    private String description;
    private String players;
    private String duration;
    private String published;
    private String age;
    private float price;
    private int rating;

    @NotNull
    @ManyToOne(optional = false)
    @JoinColumn(name = "editor",referencedColumnName = "id")
    private Editor editor;

    @ManyToMany(fetch= FetchType.LAZY)
    @JoinTable(	name = "game_author",
                joinColumns = @JoinColumn(name = "game_id"),
                inverseJoinColumns = @JoinColumn(name = "author_id"))
    private Set<Author> author = new HashSet<>();

    @ManyToMany(fetch= FetchType.LAZY)
    @JoinTable(	name = "game_category",
                joinColumns = @JoinColumn(name = "game_id"),
                inverseJoinColumns = @JoinColumn(name = "category_id"))
    private Set<Category> categories = new HashSet<>();

    @ManyToMany(fetch= FetchType.LAZY)
    @JoinTable(	name = "expansion",
                joinColumns = @JoinColumn(name = "base_game_id"),
                inverseJoinColumns = @JoinColumn(name = "expansion_id"))
    private Set<Game> expansions = new HashSet<>();

    public Long getId() {
        return id;
    }
    public void setId(Long id) {
        this.id = id;
    }
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
    public String getImage() {
        return image;
    }
    public void setImage(String image) {
        this.image = image;
    }
    public String getDescription() {
        return description;
    }
    public void setDescription(String description) {
        this.description = description;
    }
    public String getPlayers() {
        return players;
    }
    public void setPlayers(String players) {
        this.players = players;
    }
    public String getDuration() {
        return duration;
    }
    public void setDuration(String duration) {
        this.duration = duration;
    }
    public String getPublished() {
        return published;
    }
    public void setPublished(String published) {
        this.published = published;
    }
    public String getAge() {
        return age;
    }
    public void setAge(String age) {
        this.age = age;
    }
    public float getPrice() {
        return price;
    }
    public void setPrice(float price) {
        this.price = price;
    }
    public int getRating() {
        return rating;
    }
    public void setRating(int rating) {
        this.rating = rating;
    }

    public Editor getEditor() {
        return editor;
    }
    public void setEditor(Editor editor) {
        this.editor = editor;
    }
    public Set<Author> getAuthor() {
        return author;
    }
    public void setAuthor(Set<Author> author) {
        this.author = author;
    }
    public Set<Category> getCategories() {
        return categories;
    }
    public void setCategories(Set<Category> categories) {
        this.categories = categories;
    }

    public Set<Game> getExpansions() {
        return expansions;
    }
    public void setExpansions(Set<Game> expansions) {
        this.expansions = expansions;
    }

}
